### DISCUSSION TITLE
Regarding the ASN files

### DISCUSSION THREAD

**Discussion by panyogesh** [aligungr/UERANSIM#312] [General]
Hi,
  Just wanted to know which ASN file and the generation script being used in UERANSIM.
  When I am doing the test the decoder fails to decode INitiateUE message. So just wanted to see how I can bridge the gap.

Thanks,
Yogesh

**Comment by aligungr** [comment]
Hi Yogesh

ASN files are located at https://github.com/aligungr/UERANSIM/tree/master/tools

NGAP files were generated with this command:

`asn1c -fcompound-names -pdu=all -findirect-choice -fno-include-deps -gen-PER -no-gen-OER -no-gen-example -D ngap`

and RRC files were generated with this command:

`asn1c -fcompound-names -pdu=all -findirect-choice -fno-include-deps -gen-PER -no-gen-OER -no-gen-example -D rrc`

Generated ASN sources and other utilities are present in https://github.com/aligungr/UERANSIM/tree/master/src/asn

Regards

**Reply by panyogesh** [reply]
Thanks...that's very helpful
